src/view/dataproviders/resourceprovider.ts

```diff
@@ -0,0 +1,68 @@
+"use strict";
+
+import {DataProvider} from "./dataprovider";
+import {TreeItem, TreeDataProvider, ProviderResult, Event, commands, Disposable, EventEmitter} from "vscode";
+import {DataProviderEnum} from "../providermanager";
+import {injectable} from "inversify";
+import {CheckInInfo} from "../../bll/entities/checkininfo";
+import {CvsResource} from "../../bll/entities/cvsresources/cvsresource";
+import {Logger} from "../../bll/utils/logger";
+
+@injectable()
+export class ResourceProvider extends DataProvider {
```

Comment:
Maybe it would be better to rename it into something like ChangesProvider? For me ResourceProvider does not provider enough info about context of this class.